cogs/conversion.py

```diff
@@ -28,60 +45,56 @@ async def on_message(self, message: discord.Message) -> None:
             message (discord.Message): The message sent by a user.
 
         """
+
         if message.author == self.bot.user:
             return
 
-        content: str = message.content
-
-        height_matches: List[Tuple[str, str]] = ConversionCog.HEIGHT_PATTERN.findall(content)
-        weight_matches: List[str] = ConversionCog.WEIGHT_PATTERN.findall(content)
-
         # Converts the imperial units to metric units
-        converted_content: str = self.convert_height_to_cm(
-            self.convert_lbs_to_kgs(content, weight_matches), height_matches
+        converted_heights = ConversionCog.HEIGHT_PATTERN.sub(
+            self.convert_height_to_cm, message.content
+        )
+        converted_weights = ConversionCog.WEIGHT_PATTERN.sub(
+            self.convert_weight_to_kg, converted_heights
         )
 
-        # Send the converted message back to the channel
-        if converted_content != content:
-            await message.channel.send(f"```{converted_content}```")
+        msg_has_converted = converted_weights != message.content
+        if msg_has_converted:
+            await message.channel.send(f"```{converted_weights}```")
 
-    def convert_height_to_cm(self, content: str, height_matches: List[str]) -> str:
+    def convert_height_to_cm(self, match: re.Match) -> str:
         """
-        Convert height values from feet and inches to centimeters (cm)
+        Regex callback function that converts each height match into a centimeter value.
+        This returns the height converted string.
 
         Args:
-            content (str): The message content sent by the user containing possible feet and inches
-            height_matches (): Regex matches that have been identified by self.height_regex
-        """
-        converted_content: str = content
-        for feet, inches in height_matches:
-            total_inches: int = int(feet) * 12 + int(inches)
-            cm: int = round(total_inches * 2.54)
-            converted_height: str = f"{cm} cm"
-            converted_content: str = ConversionCog.HEIGHT_PATTERN.sub(
-                converted_height, converted_content
-            )
-
-        return converted_content
-
-    def convert_lbs_to_kgs(self, content: str, weight_matches: List[str]) -> str:
+            match (re.Match): The match found by the height regex.
         """
-        Convert weight values from pounds to kilograms (kg)
 
+        assert match.group(1), f"Expected {match.group(1)} to be not None"
```

Comment:
If this will go wrong it will say "Expected None not to be None".

Reply:
Please review the new message to see if that is informative enough.